src/toil/jobStores/aws/jobStore.py

```diff
@@ -1158,3 +1121,84 @@ def toItem(self):
+                                                         src_key_name=srcKey.name,
+                                                         src_version_id=srcKey.version_id,
+                                                         part_num=partIndex + 1,
+                                                         start=partIndex * partSize,
+                                                         end=min(partIndex * partSize + partSize,
+                                                                 totalSize) - 1,
+                                                         headers=headers)
+                log.debug('Successfully copied part %d of %d' % (partIndex, totalParts))
+
+            except Exception as e:
+                if partIndex <= 4:
+                    exceptions[partIndex] = traceback.format_exc()
+                else:
+                    exceptions[partIndex] = e
+                log.debug('Part number %s of copy failed.', partIndex)
+                return None
+            else:
+                return part
+
+    partSize = defaultPartSize
+    totalSize = srcKey.size
+    totalParts = (totalSize + partSize - 1) / partSize
+    exceptions = collections.OrderedDict()
+    # We need a location-agnostic connection to S3 so we can't use the one that we
+    # normally use for interacting with the job store bucket.
+    with closing(boto.connect_s3()) as s3:
+        for attempt in retry_s3():
+            with attempt:
+                dstBucket = s3.get_bucket(dstBucketName)
+                upload = dstBucket.initiate_multipart_upload(dstKeyName, headers=headers)
+        log.info("Initiated multipart copy from 's3://%s/%s' to 's3://%s/%s'." %
+                 (srcKey.bucket.name, srcKey.name, dstBucketName, dstKeyName))
+        try:
+            with ThreadPoolExecutor(max_workers=min(cpu_count() * 16, totalParts, 128)) as executor:
+                finishedParts = set(executor.map(_copyPart, range(0, totalParts)))
+                if exceptions:
+                    for partIndex, tb in exceptions[0:4].items():
+                        log.info("Part %d of copy failed with the following exception:\n%s" %
+                                 (partIndex, tb))
+                    for partIndex, exception in exceptions[4:].items():
+                        log.debug("Part %d of copy failed with the following exception:\n%s" %
+                                  (partIndex, exception))
+                    raise RuntimeError('One or more parts failed to copy. See log for details.')
+                assert len(finishedParts) == totalParts
+        except:
+            with panic(log=log):
+                upload.cancel_upload()
+        else:
+            for attempt in retry_s3():
+                with attempt:
+                    try:
+                        return upload.complete_upload()
+                    finally:
+                        log.info("Completed copy from 's3://%s/%s' to 's3://%s/%s'." %
```

Comment:
This shouldn't be in a finally or you'll get the the log line on every failed (!) attempt.

Common, Jake, these are stupid beginner mistakes.